tests/pool_alloc_corruption.c

```diff
@@ -0,0 +1,49 @@
+#include <pool_alloc.h>
+#include <stdc.h>
+#include <malloc.h>
+#include <ktest.h>
+#include <common.h>
+
+void another_int_constr(void *buf, __unused size_t size) {
+  int *num = buf;
+  *num = 0;
+}
+void another_int_destr(__unused void *buf, __unused size_t size) {
+}
+
+int test_pool_alloc_corruption() {
```

Comment:
You can avoid code duplication by merging this function into one: `test_pool_alloc(int corrupt_memory)` and then using both variants from proxy functions:
```
int test_pool_alloc_corruption(){
  return test_pool_alloc(1);
} ... etc.
```